doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,401 @@
+       +=========+=========+=========+                          +---------------+       |
+       | Header  | ...     | ...     |                          | OID           |<-----------------+
+       +---------+---------+---------+                          | Num Chunk     |       |          |
+    +--| Entry0  | ...     | ...     |                          | <Offset, len> |       |          |
+    |  +---------+---------+---------+                          | <Offset, len> |-------|----+     |
+    |  | ...     | ...     | ...     |                          | ...           |       |    |     |
+    |  +---------+---------+---------+                          +---------------+       |    |     |
+    |  | EntryK  | ...     | ...     |                                                  |    |     |
+    |  +---------+---------+---------+                                                  |    |     |
+    |                                                                                   |    |     |
+    |                                                                                   |    |     |
+    |  +---------------+  +----------+  +-------------+                                 v    v     |
+    +->| leafnode      |  | leafnode |  | leafnode    |               +------------+------------+  |
+       +===============+  +==========+  +=============+               | Block0     | Block1     |  |
+       | OID           |  | OID      |  | OID         |               +============+============+  |
+       | Block no      |  | Block no |  | Block no    |               | Free       |            |  |
+       | Offset        |  | Offset   |  | Offset      |               +------------+            +  |
+       | Extent        |  | Extent   |  | Extent      |----------+    | Small data | Data       |  |
+       +---------------+  +----------+  +-------------+          |    +------------+            +  |
+                                                                 +-+->| Extent     |            |  |
+                                                                      +------------+------------+  |
+                                                                          |                        |
+                                                                          +------------------------+
+
+
+WAL
+---
+Each AG has a WAL.
+The datas written to the WAL are all metadta updates, free space update and data.
+Note that only data smaller than the predefined threshold needs to be written to the WAL.
+The larger data is written to the unallocated free space and its onode's extent_map is updated accordingly 
+(also on-disk extent map). We statically allocate WAL partition aside from data partition pre-configured.
+
+
+Partition
+---------
+Initially, PoseidonStore employs static allocation of partition. The number of sharded partitions
+is fixed and the size of each partition also should be configured before running cluster.
```

Comment:
I *think* you mean here that each reactor thread gets a static set of AGs.  You've gone to the trouble of introducing crimson reactor threads and AGs in this document, please connect those concepts to "partition" here.